corehq/apps/app_manager/dbaccessors.py

```diff
@@ -71,6 +71,7 @@ def get_latest_released_app_versions_by_app_id(domain):
     return {r['key'][2]: r['key'][3] for r in results}
 
 
+@quickcache(['domain', 'app_id'])
```

Comment:
I think you want a much longer timeout here, the default for quickcache is quite short